tests/integration/cairo1/hint_locals/deploy/deploy_test.cairo

```diff
@@ -135,14 +103,9 @@ fn test_deploy_with_ctor_obsolete_calldata() {
     let class_hash = declare('minimal').unwrap();
     assert(class_hash != 0, 'declared class_hash != 0');
 
-    let prepare_result = prepare(class_hash, constructor_calldata).unwrap();
+    let prepare_result = prepare(class_hash, @constructor_calldata).unwrap();
 
-    let prepared_contract = PreparedContract {
-        contract_address: prepare_result.contract_address,
-        class_hash: prepare_result.class_hash,
-        constructor_calldata: prepare_result.constructor_calldata
-    };
-    match deploy(prepared_contract) {
+    match deploy(prepare_result) {
```

Comment:
maybe we could add `test_deploy_doesnt_move_calldata` test case at the end of this test suite?

Reply:
We should